prime-router/src/main/kotlin/common/DateUtilities.kt

```diff
@@ -0,0 +1,447 @@
+     * supposed to be represented by `-0000`, only ever `+0000`. That said, RFC3339 does
+     * offer the opportunity to use `-0000` to reflect an "unknown offset time", so it
+     * is still valid and does parse. Also, Java understands and can parse from `-0000`
+     * to `+0000`, so we are not breaking our implementation there.
+     *
+     * In addition to RFC3339 allowing for `-0000`, the HL7 spec allows for that value too,
+     * so we should be good in a system that is HL7 compliant.
+     *
+     * RFC Link: https://datatracker.ietf.org/doc/html/rfc3339#section-4.3
+     */
+    fun convertPositiveOffsetToNegativeOffset(value: String): String {
+        // look for the +0 offset
+        val re = Regex(".+?\\+(00|0000|00:00)$")
+        val match = re.find(value)
+        // check to see if there is a match, if there isn't return the date as expected
+        return when (match?.groups?.isNotEmpty()) {
+            true -> {
+                // get the offset value at the end of the string
+                val offsetValue = match.groups.last()?.value
+                // if there's actually a match, and all of the values in the offset are zero
+                // because we only want to do this conversion IF the offset is zero. we never
+                // want to do this if the offset is some other value
+                if (offsetValue != null && offsetValue.all { it == '0' || it == ':' }) {
+                    // create our replacement values
+                    // I am doing it this way because it's possible that our desired level of
+                    // precision for date time offset could change. I don't want my code to
+                    // assume that it will always be +0000. +00 and +00:00 are also acceptable values,
+                    // so I want this to be able to handle those options as well
+                    val searchValue = "+$offsetValue"
+                    val replaceValue = "-$offsetValue"
+                    // replace the positive offset with the negative offset
+                    value.replace(searchValue, replaceValue)
+                } else {
+                    // we had an offset, but it's not what we expected, so just return the
+                    // original value we were passed in to be safe
+                    value
+                }
+            }
+            // the regex didn't match, so return the original value we passed in
+            else -> value
+        }
+    }
+
+    /**
+     * Given a [dateTimeValue] and a [report] object, this will attempt to format the date for a
+     * receiver, by doing some checking on what is set for the receiver of the report. It also accepts
+     * a [dateTimeFormat] that it will use first if it exists, and then the one from the [report]
+     * object, and then finally default to the [DateTimeFormat.OFFSET] value.
+     */
+    fun formatDateForReceiver(
```

Comment:
Here I have a separation of concerns issue. Common and utilities should be very generic and useful in multiple contexts. Here you are adding very ReportStream specific methods. I suggest this isn't necessary since `dateTimeFormat` is passed in.